### PR TITLE
DRAFT: block rpcs and explorer interface

### PR DESCRIPTION
**Pull request by maxtori**
- need to split in several commits
-  wasmer tests are not passing (probably due to wasm-bindgen dependency)
- operations and messages not shown (due to tokio fs)

### CODE REVIEW

**Review by afck — COMMENTED** [review]
Looks great so far! :+1:

**Comment by maxtori** [comment]
thanks for the remarks, I will correct/clean and continue the interface

File: linera-explorer/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+[package]
+name = "linera-explorer"
+version = "0.1.0"
+edition = "2021"
+description = "Block explorer for linera chain"
```
[the remark this answers is not in the document]
Comment (ma2bd) [reply linera-explorer/Cargo.toml:5]: "Linera protocol" as in the whitepaper

File: linera-storage/src/lib.rs [hunk 2]
```diff
@@ -282,6 +289,24 @@ where
         Ok(value.with_hash_unchecked(hash))
     }
 
+    async fn read_values(
+        &self,
+        from: CryptoHash,
+        limit: u32,
+    ) -> Result<Vec<HashedValue>, ViewError> {
+        let mut limit_ref = limit;
```
[the remark this answers is not in the document]
Comment (ma2bd) [reply linera-storage/src/lib.rs:297]: `mut limit` in the parameters?

**Pull request closed without merging**